apps/site/pages/docs/api-extensions/extending-api-schema.mdx

```diff
@@ -0,0 +1,378 @@
+
+#### Step 3 - Creating resolvers
+
+Now, you'll create a resolver to define how the new `customData` field should be resolved, or in other words, what data or logic should be associated with them when queried.
+
+1. Create a new `<resolver>.graphql` file inside the `vtex/resolvers` folder. Let's continue the [StoreProduct](https://v1.faststore.dev/reference/api/objects#storeproduct) type
+example, so the name of the file is `product.ts`.
+
+<Callout type="info" emoji="ℹ️">
+  The way that it will be organized inside this folder is flexible, it's possible to create multiple resolvers files or group them all together in just one file.
+</Callout>
+
+2. Define the resolver in the `product.ts` file. This resolver specifies how to fetch the data for the `customData` field based on the root object, which represents the `StoreProduct`.
+
+```ts copy filename="graphql/vtex/resolvers/product.ts"
+
+import type { StoreProductRoot } from '@faststore/api'
+
+const productResolver = {
+  StoreProduct: {
+    customData: (root: StoreProductRoot) => {
+      return 'My item id: ' + root.itemId
+    },
+  },
+}
+
+export default productResolver
+```
+
+Note that you can leverage the use of TypeScript by typing the root param as [StoreProductRoot](https://v1.faststore.dev/reference/api/root-object-fields#root-product):
+
+![using-root-field-type](https://vtexhelp.vtexassets.com/assets/docs/src/root-field-ts___3a10ec8a1aeba9770c7ee982995d55bc.gif)
+
+3. Open the `index.ts` file inside the `vtex/resolvers` folder and import the resolver, `StoreProductResolver`, you created in the previous step. 
+
+4. Create a `resolvers` object that includes the imported resolver(s).
+
+```ts copy filename="graphql/vtex/resolvers/index.ts"
+
+import { default as StoreProductResolver } from './product'
+
+const resolvers = {
+  ...StoreProductResolver,
+}
+
+export default resolvers
+```
+
+Once you have defined these files in the `vtex` folder, the new fields are already available to use. 
+
+<Callout type="warning" emoji="⚠️">
+  If the changes you made still not available, refer to the troubleshooting section [GraphQL changes not visible during development](/docs/api-extension/troubleshooting#graphql-changes-not-visible-during-development).
+</Callout>
+ 
+
+## Extending third-party API schemas
+
+As stores tend to grow, so does the possibility of consuming new data that is not provided by default in the FastStore API or other VTEX APIs. 
+As such, FastStore must consume new data from third-party APIs.
+
+### Step by step
+#### Step 1- Preparing the folders and files
+
+<Callout type="warning" emoji="⚠️">
+  If you already have a directory structure created in the [Extending VTEX API Schemas](/docs/api-extension/extending--api-schema#step-1---preparing-the-folders-and-files),
+  for third-party API schemas you only need to add the `thirdParty` folder inside `src/graphql`.
```

Comment:
```suggestion
  For third-party API schemas you only need to add the `thirdParty` folder inside `src/graphql`.
```

I'd re-frase it so users know they shouldn't use the Extending VTEX API Schemas directory, but I wouldn't put it as a condition. They'll need the `thirdParty` folder anyways.